va/va.go

```diff
@@ -327,15 +372,35 @@ func (va *ValidationAuthorityImpl) validateTLSSNI01WithZName(ctx context.Context
 		},
 	}
 	if problem != nil {
-		return validationRecords, problem
+		return nil, validationRecords, problem
 	}
+	thisRecord := &validationRecords[0]
 
 	// Make a connection with SNI = nonceName
 	portString := strconv.Itoa(va.tlsPort)
 	hostPort := net.JoinHostPort(addr.String(), portString)
-	validationRecords[0].Port = portString
+	thisRecord.Port = portString
 
 	certs, problem := va.getTLSSNICerts(hostPort, identifier, challenge, zName)
+
+	// If there was a problem getting the certificates and the IPv6 first feature
+	// is enabled, then try to fetch the certificates again using a fallback
+	// address
+	if problem != nil && features.Enabled(features.IPv6First) {
```

Comment:
I'm open to that - would that be a fair compromise to using the dialer directly? It will still be some duplication but I haven't been able to think of a way to avoid that yet.